frontend/app/components/input-checkboxes.tsx

```diff
@@ -0,0 +1,72 @@
+import type { ComponentProps, ReactNode } from 'react';
+
+import { InputCheckbox } from './input-checkbox';
+import { InputError } from './input-error';
+import { InputHelp } from './input-help';
+import { InputLegend } from '~/components/input-legend';
+import { cn } from '~/utils/tw-utils';
+
+export interface InputCheckboxesProps {
+  errorMessage?: string;
+  helpMessagePrimary?: React.ReactNode;
+  helpMessagePrimaryClassName?: string;
+  helpMessageSecondary?: React.ReactNode;
+  helpMessageSecondaryClassName?: string;
+  id: string;
+  options: Omit<ComponentProps<typeof InputCheckbox>, 'aria-describedby' | 'aria-errormessage' | 'aria-invalid' | 'aria-required' | 'id' | 'name' | 'required'>[];
+  legend: ReactNode;
+  name: string;
+  required?: boolean;
+}
+
+const InputCheckboxes = ({ errorMessage, helpMessagePrimary, helpMessagePrimaryClassName, helpMessageSecondary, helpMessageSecondaryClassName, id, legend, name, options, required }: InputCheckboxesProps) => {
+  const inputErrorId = `input-checkboxes-${id}-error`;
+  const inputHelpMessagePrimaryId = `input-checkboxes-${id}-help-primary`;
+  const inputHelpMessageSecondaryId = `input-checkboxes-${id}-help-secondary`;
+  const inputLegendId = `input-checkboxes-${id}-legend`;
+  const inputWrapperId = `input-checkboxes-${id}`;
+
+  function getAriaDescribedby() {
+    const ariaDescribedby = [inputLegendId];
+    if (helpMessagePrimary) ariaDescribedby.push(inputHelpMessagePrimaryId);
+    if (helpMessageSecondary) ariaDescribedby.push(inputHelpMessageSecondaryId);
+    return ariaDescribedby.length > 0 ? ariaDescribedby.join(' ') : undefined;
+  }
+
+  return (
+    <fieldset id={inputWrapperId} data-testid={inputWrapperId}>
+      <InputLegend id={inputLegendId} required={required} className="mb-2">
+        {legend}
+      </InputLegend>
+      {helpMessagePrimary && (
+        <InputHelp id={inputHelpMessagePrimaryId} className={cn('mb-2', helpMessagePrimaryClassName)} data-testid="input-field-help-primary">
+          {helpMessagePrimary}
+        </InputHelp>
+      )}
+      <ul className="space-y-2">
+        {options.map((option, index) => {
+          const inputCheckboxId = `${id}-option-${index}`;
+          return (
+            <li key={inputCheckboxId}>
```

Comment:
You should use something else for the key that isn't dependent on the index. Is there another property of the option that is unique and non-volatile?

See <https://react.dev/learn/rendering-lists#why-does-react-need-keys>

![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/48123208/15b02600-9aae-4cff-b085-72f175e1d43a)

Reply:
If you're wondering when this could actually be a problem in real life, consider a page that sorts the `options` differently depending on if the language is english vs french (think a province list that's sorted alphabetically on the page).

Reply:
Sure, that makes sense.

Implemented in commit dec2ceef9218bed7957453837c8f49c7cb1d2692.